x/farm/spec/01_concepts.md

```diff
@@ -0,0 +1,13 @@
+<!--
+order: 1
+-->
+
+# Concept
+
+**Disclaimer: This is work in progress. Mechanisms are susceptible to change.**
+
+The `x/farm` module is an inter-module accessible, generalized yield farming reward distribution module based onf the [F1 fee distribution module](https://github.com/cosmos/cosmos-sdk/tree/master/docs/spec/fee_distribution).
```

Comment:
Is the idea that the calling module will execute a "distribute rewards" method, and setup "reard recipients" ?

Reply:
The current expected usage flow is that when a module needs to set up a pool to distribute reward, it will use `NewFarm` to create a new farm. Then, if rewards are minted (whether it’s from inflation or fees), it would use the `AllocateAssetToFarm` method to deposit the asset to the farm.

Also, `DepositShareToFarm`, `WithdrawShareFromFarm`, and `WithdrawRewardsFromFarm` method allows the users to deposit or claim rewards, but these three methods return the `rewards sdk.Coins` that the user should receive. However, it only returns the the rewards that the user should receive, but doesn’t actually transfer the assets to the user–as this is the responsibility of the module that is calling the farm module. Also, for the `DepositShareToFarm`, `WithdrawShareFromFarm` methods, it is responsibility of the module calling the x/farm module to verify the assets relating to a specific share, and moving these assets to the correct module.